packages/x-components/src/components/page-selector.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+  <nav class="x-flex x-items-center x-justify-center x-gap-2" aria-label="Pagination">
+    <button
+      @click="selectPage(currentPage - 1)"
+      class="x-button x-button-lead"
+      :disabled="currentPage === 1"
+      data-test="previous-page-button"
+    >
+      <slot name="previous-page-button">Prev</slot>
+    </button>
+
+    <button
+      v-for="page in visiblePages"
+      :key="page"
+      @click="selectPage(page)"
+      class="x-button"
+      :class="{
+        'x-button-lead x-cursor-default': page === currentPage,
+        'x-button-ghost': page !== currentPage,
+        'x-cursor-not-allowed': page === '...'
+      }"
```

Comment:
We need to think about how can we style this from the consumer side. Maybe it's a good idea to have selected and disabled states so we can target them from the outside